word2vec.py

```diff
@@ -0,0 +1,138 @@
+from gensim.models import Word2Vec, KeyedVectors
+from sklearn.metrics import roc_curve, roc_auc_score, auc
+import pandas as pd
+import numpy as np
+import random
+import os
+import matplotlib.pyplot as plt
+
+
+class Sentences(object):
+    """
+    Extracts title + abstracts from Pubtator data. Replaces any instance of a 
+    gene or disease with the Entrez gene ID or MESH ID, respectively. 
+    """
+    def __init__(self, filename):
+        self.filename = filename
+ 
+    def __iter__(self):
+        curr_pmid = curr_title = curr_abstract = curr_total = None
+        for line in open(self.filename):
+            if "|t|" in line:
+                curr_title = line.split("|")[2]
+                continue
+            if "|a|" in line:
+                if curr_total is not None:
+                    yield curr_total.split()
+                curr_pmid = line.split("|")[0]
+                curr_abstract = line.split("|")[2]
+                if curr_title is not None:
+                    curr_total = curr_title + curr_abstract  # combine title and abstract
+                continue
+            else:
+                if curr_pmid is not None and curr_total is not None:
+                    if "Disease" in line or "Gene" in line:  # targeting gene-disease pairs
+                        features = line.split("\t")
+                        if int(features[0]) == int(curr_pmid) and features[5].strip() != "":
+                            curr_total = curr_total.replace(features[3], features[5].strip(), 1)
+        yield curr_total.split()
+                    
+
+def create_word2vec(sentences):
+    """
+    Creates a word2vec model.
+    @param sentences: list of list of words in each sentence (title + abstract).
+    """
+    model = Word2Vec(sentences, size=500, window=5, min_count=1, workers=4)
+    model.save("word2vec.model")
+    return model
+
+
+def get_gene_disease_pairs(gene_disease_filename, do_mesh_filename):
+    """
+    Extracts hetionet gene-disease pairs and generates negative pairs by randomizing positive pairs. 
+    @param gene_disease_filename: file containing hetionet gene disease pairs
+    @param do_mesh_filename: file containing corresponding doid and mesh ids (because hetnet
+        contains only doids)
+    """
+    random.seed(100)  # reproducibility
+    gene_disease_df = pd.read_csv(gene_disease_filename, sep="\t")
+    do_mesh_df = pd.read_csv(do_mesh_filename, sep="\t")
+
+    # create doid-mesh list
+    do_mesh_pairs = dict(zip(do_mesh_df.doid_code, "MESH:"+do_mesh_df.mesh_id))
```

Comment:
Would make your code easier to read if you convert long lines into lines with spaces:
```python
do_mesh_pairs = dict(zip(
     do_mesh_df.doid_code,
     "MESH:"+do_mesh_df.mesh_id
))
```